safe/gis/expressions.py

```diff
@@ -0,0 +1,37 @@
+# coding=utf-8
+from qgis.core import (
+    qgsfunction,
+    QgsMapLayerRegistry,
+    QgsExpressionContextUtils,
+)
+from safe.definitions.provenance import provenance_layer_analysis_impacted_id
+
+__copyright__ = "Copyright 2016, The InaSAFE Project"
+__license__ = "GPL version 3"
+__email__ = "info@inasafe.org"
+__revision__ = '$Format:%H$'
+
+
+@qgsfunction(
+    args='auto', group='InaSAFE', usesGeometry=False, referencedColumns=[])
+def inasafe_impact_analysis_layer(field, feature, parent):
+    """Retrieve a value from a field in the impact analysis layer."""
+    _ = feature, parent  # NOQA
+    project_context_scope = QgsExpressionContextUtils.projectScope()
+    registry = QgsMapLayerRegistry.instance()
+
+    key = provenance_layer_analysis_impacted_id['provenance_key']
```

Comment:
so the purpose is this expression alaso can be used by the user? if so, yes I agree to keep it simple, but it means we need to create another function if we want to get the data from another layer, is it correct?